src/Botan.php

```diff
@@ -181,33 +227,37 @@ public static function shortenUrl($url, $user_id)
         }
 
         $cached = BotanDB::selectShortUrl($user_id, $url);
-        if (!empty($cached[0]['short_url'])) {
-            return $cached[0]['short_url'];
-        }
 
-        $request = str_replace(
-            ['#TOKEN', '#UID', '#URL'],
-            [self::$token, $user_id, urlencode($url)],
-            self::$shortener_url
-        );
+        if (!empty($cached)) {
+            return $cached;
+        }
 
-        $options = [
-            'http' => [
-                'ignore_errors' => true,
-                'timeout'       => 3,
-            ],
-        ];
+        try {
+            $response = self::$client->request(
+                'POST',
+                str_replace(
+                    ['#TOKEN', '#UID', '#URL'],
+                    [self::$token, $user_id, urlencode($url)],
+                    '/s/?token=#TOKEN&user_ids=#UID&url=#URL'
+                )
+            );
 
-        $context  = stream_context_create($options);
-        $response = @file_get_contents($request, false, $context);
+            $response = (string) $response->getBody();
+        } catch (RequestException $e) {
+            $response = ($e->getResponse()) ? (string) $e->getResponse()->getBody() : '';
+        } finally {
+            if (!filter_var($response, FILTER_VALIDATE_URL) === false) {
```

Comment:
Just using `filter_var($response, FILTER_VALIDATE_URL) !== false` is easier to read.